airbyte-integrations/connectors/source-datadog/source_datadog/source.py

```diff
@@ -43,13 +54,39 @@ def streams(self, config: Mapping[str, Any]) -> List[Stream]:
             Users(**args),
         ]
 
+        # Get the list of queries from the config
+        queries = config.get("queries", [])
+
+        # Create a stream for each query in the list
+        query_streams = []
+        for query in queries:
+            name = query["name"]
+            data_source = query["data_source"]
+            query_string = query["query"]
+
+            # Create a new stream using the query name, data source, and query string
+            new_stream = SeriesStream(
+                name=name,
+                data_source=data_source,
+                query_string=query_string,
+                **args,
+            )
+            query_streams.append(new_stream)
+
+        # Combine the base streams and query streams
+        return base_streams + query_streams
```

Comment:
There a lot of logic happening here. Please add unit test to validate that.
What happen if you send a invalid query?

Reply:
Added unit tests and validations as well.
In reality, invalid query won't be the scenario - If the query field is empty or invalid, Datadog timeseries API returns an empty response following same response schema. The _queries_ is optional but the fields _name_, _datasource_ and _query_ are all mandatory. 
<img width="839" alt="Screenshot 2023-06-12 at 18 58 41" src="https://github.com/airbytehq/airbyte/assets/99747757/a95b5f3c-cbfd-454d-a2c7-5220e7cd9342">